rubin_sim/scheduler/surveys/field_survey.py

```diff
@@ -0,0 +1,71 @@
+import numpy as np
+
+from rubin_sim.utils import _raDec2Hpid
+
+from .dd_surveys import Deep_drilling_survey
```

Comment:
Seems like we should probably make a more general "user defined scripted survey" class to inherit from, then make that available for this and for DD surveys, and potentially also ToO and star field kind of surveys. Could make a meta-list of what kinds of things people might generally need to customize (filter sequence, locations of fields, numbers of visits and exposure times per visit (per filter perhaps, could expand this to a list matching the filter sequence), basis functions and detailers, other info needed for programming sequences (filter change time, read time, etc) and then let that be the framework for further customization?

Reply:
I think that is a good idea..